### ISSUE TITLE
CassandraDatacenter spec.podTemplateSpec.spec.serviceAccountName is not passed to StatefulSet

### ISSUE THREAD

**Issue by apa64** [k8ssandra/cass-operator#523]
### What happened?

I am deploying Cassandra with cass-operator to OpenShift 4.12. By default OpenShift does not allow running containers with fixed UIDs so I am working around that by creating a SecurityContextConstraint.I need to set ServiceAccount for Cassandra pods so that the SCC is active for the pods and allows the UID 999 in pod.

I set the serviceAccountName for the pods in CassDC CRD configuration like this (unrelated parts left out):

```yaml
apiVersion: cassandra.datastax.com/v1beta1
kind: CassandraDatacenter
spec:
  podTemplateSpec:
    spec:
      serviceAccountName: run-as-nonroot
```

The SA in StatefulSet remains `default` (unrelated parts left out):

```yaml
kind: StatefulSet
spec:
  template:
    spec:
      serviceAccount: default
      serviceAccountName: default
```

### What did you expect to happen?

I expected the serviceAccountName in STS to be the same as in the CassDC.

### How can we reproduce it (as minimally and precisely as possible)?

Set the serviceAccountName in podTemplateSpec in the CassDC configuration.

### cass-operator version

1.14.0

### Kubernetes version

Kubernetes Version: v1.23.12+8a6bfe4

### Method of installation

ArgoCD (GitOps)

### Anything else we need to know?

Possibly related issues:
- https://github.com/k8ssandra/cass-operator/issues/264
- https://github.com/k8ssandra/cass-operator/issues/518
- https://github.com/k8ssandra/k8ssandra/issues/1154

**Comment by burmanm** [comment]
Hey,

Use `CassandraDatacenter.Spec.ServiceAccount`, that should pass it to the StatefulSet.

**Comment by apa64** [comment]
Thank you @burmanm, this was the solution! The StatefulSet now has:

```
      serviceAccount: run-as-nonroot
      serviceAccountName: run-as-nonroot
```

> Use `CassandraDatacenter.Spec.ServiceAccount`, that should pass it to the StatefulSet.

Is it possible to fix the documentation for this parameter? I followed these two trying to find the right place for the setting:
- https://github.com/k8ssandra/cass-operator/blob/v1.14.0/config/crd/bases/cassandra.datastax.com_cassandradatacenters.yaml#L5605
- https://docs.k8ssandra.io/reference/crd/releases/cass-operator-releases/cass-operator-crds-1.10.x/#cassandradatacenterspecpodtemplatespecspec

**Comment by burmanm** [comment]
Hey, those are autogenerated from the Kubernetes definition. I think we have somewhere in the documentation (although perhaps this should be clearer) that we do not blindly use all the information from PodTemplateSpec (there are multiple fields which we will ignore from there) and that PodTemplateSpec should not be used in almost any case. There are few exceptions, but it's more like "know what you're doing" type of situations. 

In other words, if there's a real use-case that requires PodTemplateSpec, then perhaps that would be a good ticket to improve on. In this case, there's perhaps no reason to let you not use the PodTemplateSpec one, but there's a possibility of conflicting options. Maybe I'll modify this behavior a bit to let both work, but with ordering..

**Comment by apa64** [comment]
Yeah, as it turned out I did not need PodTemplateSpec and yes, setting ServiceAccount in there would conflict the other one. Actually I just found documentation for serviceAccount: https://github.com/k8ssandra/cass-operator/blob/master/operator/example-cassdc-yaml/cassandra-3.11.x/example-cassdc-full.yaml#L119

As it is not planned to use PodTemplateSpec I guess this issue can be closed?

**Comment by burmanm** [comment]
Leave this open for now.. if I'll make a small change to the behavior, I'll refer this ticket. If not, I can close this later.
